### PR TITLE
[DO NOT Review] os/drivers/input : initial touch driver Test code

### PR DESCRIPTION
**Pull request by anjana348**
Initial ist415 touch driver for 4.3 inch(480*800) lcd
sample app update for touch test
for ex -
xres : 480, yres:800
touch_interrupt: Touch interruptttttt...
corrdinates are id 0 status 1 type 0 x : 627 y : 157

 coordinates x : 627 y : 157
touch_interrupt: Touch interruptttttt...
corrdinates are id 0 status 1 type 0 x : 315 y : 221

 coordinates x : 315 y : 221
touch_interrupt: Touch interruptttttt...
corrdinates are id 0 status 1 type 0 x : 325 y : 286

 coordinates x : 325 y : 286

### CODE REVIEW

File: os/drivers/input/ist415.c [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+
+  outlen = sizeof(struct touch_sample_s);
+  if (priv->int_pending)
+    {
+      ret = ist415_get_touch_data(priv, buffer);
+    }
+//printf("not pending \n");
+  /* Clear pending flag with critical section */
+	irq_disable(9);
+  flags = enter_critical_section();
```
Comment (ewoodev) [inline os/drivers/input/ist415.c:264]: let's add #include <tinyara/irq.h>

File: os/board/rtl8730e/src/rtl8730e_ist415.c [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+#define IST415_I2C_PORT		1
+
+#define IST415_I2C_FREQ		100000
+#define IST415_I2C_ADDRLEN		7
+
+#define IST415_I2C_ADDR		(0xA0 >> 1) 
+
+/*other pin config */
+#define IST415_GPIO_RESET_PIN		PA_5
+#define IST415_GPIO_I2C_PIN		PA_2
```
Comment (ewoodev) [inline os/board/rtl8730e/src/rtl8730e_ist415.c:53]:
```
#if CONFIG_RTL8730E_BOARD_REVISION >= 6
#define IST415_GPIO_I2C_PIN PA_4
#else
#define IST415_GPIO_I2C_PIN PA_2
#endif
```

Could you please apply it?

File: apps/examples/lcd_test/example_lcd.c [hunk 3]
```diff
@@ -371,6 +425,11 @@ int lcd_test_main(int argc, char *argv[])
 	int fd = 0;
 	int p = 0;
 	char port[20] = { '\0' };
+	pid_t touch = task_create("touch", SCHED_PRIORITY_DEFAULT, 8096, touch_test, NULL);
```
Comment (sunghan-chang) [inline apps/examples/lcd_test/example_lcd.c:428]: This is a LCD test example. I mean the main purpose is testing LCD. But some LCD can be touchless. Then, is this example ok to run? No need to cover new codes with touch config?

File: apps/examples/lcd_test/example_lcd.c [hunk 4]
```diff
@@ -360,6 +365,55 @@ static void test_fps(void)
...
+		}	
+		/* since we are the only reader, no need to check failure case
+		 * as we are garunteed that there is data to read by POLLIN */
+		ret = read(fd, &buf, sizeof(struct touch_sample_s));
+		if (ret != - 1) {
+			printf("\n coordinates id: %d, x : %d y : %d touch type: %d\n", buf.point[0].id, buf.point[0].x, buf.point[0].y, buf.point[0].flags);
+			if (buf.point[0].flags == TOUCH_DOWN) {
+				printf("touch press event \n");
+			}
+			else if (buf.point[0].flags == TOUCH_MOVE) {
```
Comment (sunghan-chang) [inline apps/examples/lcd_test/example_lcd.c:403]: coding rule violation

} else if ( ) {

File: apps/examples/lcd_test/example_lcd.c [hunk 5]
```diff
@@ -360,6 +365,55 @@ static void test_fps(void)
...
 }
 
+static void touch_test(void)
+{
+	/* read first 10 events */
+	char c;
+	int ret;
+
+	int fd = open("/dev/input0", O_RDONLY);
+	printf("in touch test\n");
```
Comment (sunghan-chang) [inline apps/examples/lcd_test/example_lcd.c:375]: Is this a necessary message?

File: apps/examples/lcd_test/example_lcd.c [hunk 6]
```diff
@@ -360,6 +365,55 @@ static void test_fps(void)
...
+		printf("Failed to open /dev/input0, errno : %d\n", get_errno());
+		return NULL;
+	}
+
+	struct pollfd fds[1];
+	fds[0].fd = fd;
+	fds[0].events = POLLIN;
+
+	struct touch_sample_s buf;
+	for (int i = 0; i < 100; i++) {
```
Comment (sunghan-chang) [inline apps/examples/lcd_test/example_lcd.c:386]: What's the meaning of 100? After 100 touch, this task will be automatically finished?
If yes, you already added task_delete at end of lcd test main function. So, how about changing this to infinite loop?

File: os/board/rtl8730e/src/rtl8730e_ist415.c [hunk 7]
```diff
@@ -0,0 +1,200 @@
+/****************************************************************************
+ *
+ * Copyright 2023 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline os/board/rtl8730e/src/rtl8730e_ist415.c:3]: 2024

File: os/include/tinyara/input/ist415.h [hunk 8]
```diff
@@ -0,0 +1,334 @@
+typedef signed char                             s8;
```
Comment (sunghan-chang) [inline os/include/tinyara/input/ist415.h:1]: Need the Samsung license

File: os/drivers/input/ist415.c [hunk 9]
```diff
@@ -0,0 +1,488 @@
...
+
+#define TOUCH_THRESHOLD 0
+#define EVENT_PACKET_SIZE 16
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/* IST415 Device */
+
+struct ist415_dev_s
```
Comment (sunghan-chang) [inline os/drivers/input/ist415.c:56]: should modify coding rule violations

File: os/include/tinyara/input/touchscreen.h [hunk 10]
```diff
@@ -68,8 +68,9 @@
 
 #include <tinyara/config.h>
 #include <tinyara/fs/ioctl.h>
+#include <tinyara/i2c.h>
 
-#ifdef CONFIG_INPUT
+//#ifdef CONFIG_INPUT
```
Comment (sunghan-chang) [inline os/include/tinyara/input/touchscreen.h:73]: Need to remove or change it to proper config.
Do we have a touch config?

**Comment by sunghan-chang** [comment]
@anjana348 Let's clean up and merge this PR.

**Pull request closed without merging**